### PR TITLE
coppwr: init at 1.5.1

### PR DESCRIPTION
**Pull request by RAVENz46**
## Description of changes

Add [coppwr](https://github.com/dimtpap/coppwr), Low level control GUI for the PipeWire multimedia server.
Requested #300268 .

This is my first contribution and I am not yet familiar with how to handle git, so I would be glad to get detailed advice.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by chewblacka** [comment]
Hi, congratulations on your first commit! Commit messages to nixpkgs have a special format, so the commit messages should be changed as follows.

Are you familiar with git interactive rebase? You can use it to re-order the commits and change the various commit messages. The command is `git rebase -i HEAD~n` where n is the number of commits you want to change, in your case 3, so the command would be: `git rebase -i HEAD~3`. An editor will pop up showing your 3 commits (oldest on top, newest at the bottom) with instructions on how you can change them. Re-ordering is as simple as cutting and pasting the lines. 

Basically you want the maintainer commit to go first. The commit message for that should be "maintainers: add RAVENz46". 

The "create package.nix" commit message should be "coppwr: init at 1.5.1". The "Add Cargo.lock" can either be squashed into the previous commit (since it is part of the package init), or reworded as "coppwr: add Cargo.lock". Generally commits to packages should be of the form "package-name: action taken". 

Let me know if you come across any issues.

**Comment by RAVENz46** [comment]
@chewblacka 
Huge thanks for you.
It seems I did it!

**Comment by chewblacka** [comment]
> @chewblacka Huge thanks for you. It seems I did it!

Well done! Git rebase interactive is very useful. p.s. I think the maintainers commit should go first. You can use git rebase to reorder the commits. Same command as before `git rebase -i HEAD~3`, but this time move the maintainers commit from the bottom to the top (i.e. make it the first commit).

**Comment by RAVENz46** [comment]
@chewblacka Done

**Review by chewblacka — APPROVED** [review]
Looks good to me!

Result of nixpkgs-review
1 package built:
coppwr

**Review by Aleksanaa — APPROVED** [review]
Looks okay, didn't test

**Comment by chewblacka** [comment]
To avoid commit spam, generally you want to amend existing commits with changes rather than create new commits. So in this case you could either use git rebase to move the commits you want to change to the front, then change them. Or perhaps the easier option is to just squash the last 3 commits into 1 with commit message: "coppwr: init at 1.5.1".

**Comment by Aleksanaa** [comment]
Please amend `coppwr: add Cargo.lock` into `coppwr: init at 1.5.1`.

**Comment by RAVENz46** [comment]
@chewblacka @Aleksanaa 
Done!
Is it fine?

**Comment by chewblacka** [comment]
Commits look good to me.

**Comment by Aleksanaa** [comment]
Okay! Will test now

File: pkgs/by-name/co/coppwr/package.nix [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  ];
+
+  preBuild = ''
+    mkdir -p $out/share/{applications,icons}
+    mkdir $out/metainfo
+
+    install -m 444 \
+        -D $src/assets/io.github.dimtpap.coppwr.desktop \
+        -t $out/share/applications
+    cp $src/assets/io.github.dimtpap.coppwr.metainfo.xml $out/metainfo/
```
Comment (Aleksanaa) [inline pkgs/by-name/co/coppwr/package.nix:55]:
```suggestion
    install -m 444 \
      -D $src/assets/io.github.dimtpap.coppwr.metainfo.xml \
      -t $out/share/metainfo
```

It should be in `share/metainfo` (Checked with `nix-locate` in `nix-index-database`)

image

File: pkgs/by-name/co/coppwr/package.nix [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    cp $src/assets/icon/scalable.svg $out/share/icons/io.github.dimtpap.coppwr.svg
```
Comment (Aleksanaa) [inline pkgs/by-name/co/coppwr/package.nix:56]:
```suggestion
    cp $src/assets/icon/scalable.svg $out/share/icons/scalable/apps/io.github.dimtpap.coppwr.svg
```

**Review by Aleksanaa — COMMENTED** [review]
Looks okay, just some files seem to be placed wrongly
image

File: pkgs/by-name/co/coppwr/package.nix (hunk 2 above)
Comment (RAVENz46) [reply pkgs/by-name/co/coppwr/package.nix]: Done
